src/main/kotlin/net/atos/zac/mail/model/Bronnen.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Lifely
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+package net.atos.zac.mail.model
+
+import net.atos.client.zgw.drc.model.generated.EnkelvoudigInformatieObject
+import net.atos.client.zgw.zrc.model.Zaak
+import org.flowable.task.api.TaskInfo
+
+class Bronnen private constructor(
+    val zaak: Zaak,
+    val document: EnkelvoudigInformatieObject?,
+    val taskInfo: TaskInfo?
+) {
+    class Builder {
+        private lateinit var zaak: Zaak
+        private var document: EnkelvoudigInformatieObject? = null
+        private var taskInfo: TaskInfo? = null
+
+        fun add(zaak: Zaak): Builder {
+            this.zaak = zaak
+            return this
+        }
+
+        fun add(document: EnkelvoudigInformatieObject): Builder {
+            this.document = document
+            return this
+        }
+
+        fun add(taskInfo: TaskInfo): Builder {
+            this.taskInfo = taskInfo
+            return this
+        }
+
+        fun build(): Bronnen {
+            return Bronnen(zaak, document, taskInfo)
+        }
+    }
+
+    companion object {
+        @JvmStatic
+        fun fromZaak(zaak: Zaak): Bronnen {
+            return Builder().add(zaak).build()
+        }
+    }
+}
```

Comment:
```suggestion
   }
    
   fun fromZaak(zaak: Zaak) = Builder().add(zaak).build()      
```

But probably even better is an extension function. https://www.baeldung.com/kotlin/extension-methods

Reply:
Extension functions are not static, so we use either need companion object or `BronnenKt` class usage in Java. Opted for the second option